lib/matplotlib/tests/test_legend.py

```diff
@@ -532,6 +532,30 @@ def test_empty_bar_chart_with_legend():
     plt.legend()
 
 
+def test_shadow_argument_types():
```

Comment:
I am not sure if this test is meaningful, but I can easily add an image comparison test if that would be preferable.

Reply:
At a minimum we need a `fig.canvas.draw()` to force the `draw` method to actually run!

Unfortunately I think we do need an image test here.  An obviously way this could break is that we keep the validation logic, but fall back to boolean logic in draw so all of these draw with a gray shadow.

Reply:
That makes sense. I'll also put the `ValueError` part to a separate test then.